designs/Consolidate-NuGetConfig_OnLinux.md

```diff
@@ -0,0 +1,216 @@
+NuGet will get user wide NuGet.Config file from consolidated path. 
+No warning will be shown.
+
+**2. Legacy path exist, but consolidated path does not exist.**
+
+NuGet will copy the NuGet.Config file from legacy path to consolidated path, and set up the legacy path to be a symbolic link to the consolidated path.
+NuGet will get user wide NuGet.Config file from consolidated path. 
+No warning will be shown.
+
+**3. Legacy path does not exist, but consolidated path exists.**
+NuGet will set up the legacy path to be a symbolic link to the consolidated path.
+
+**4. Both legacy path and consolidated path exist, but legacy path is not a symbolic link.**
+
+**4.1 If NuGet could determine how to consolidate the contents of the two**
+If NuGet could determine how to consolidate the contents of the two, NuGet will consolidate the two into consolidated path, and set up the legacy path to be a symbolic link to the consolidated path. After consolidation, NuGet will get config file from consolidated path.
+No warning will be shown.
+
+**4.2 If NuGet could not determine how to consolidate the contents of the two**
+If NuGet could not determine how to consolidate the contents of the two, NuGet will get user wide NuGet.Config file from **legacy path**.
+NuGet will show a warning, encouraging customers to consolidate the contents of the two and only keep the one in consolidated path. If customers finish that, the next time, it will jump to scenario 3. legacy path does not exist, but consolidated path exists, so NuGet will help them set up the legacy path to be a symbolic link to the consolidated path. 
+
+Warning: there are two user-wide NuGet.Config files. Put the wanted config file in consolidated path, delete the one from legacy path. Next time, NuGet will set up the legacy path to be a symbolic link to the consolidated path. For more details, please refer to: a link of document.
+
+#### Pros: 
+If customer uses gloabal.json to pin different versions of dotnet dynamically, the old version of dotnet will pick config file from legacy path, which is actually the same file with the one in consolidated path. So all versions of dotnet could maintain a single copy of NuGet.Config files. 
+
+#### Cons: 
+Legacy path still exists for all scenarios, even if the NuGet.Config file is only in consolidated path, legacy path will be created as a symbolic link.
+
+If customer sets legacy path wrongly by themselves(not following instructions in warning of scenario 4.2 and let NuGet to do that for them), e.g. a symbolic link pointing to a wrong path, NuGet is able to detect that, but NuGet won't know if it's on purpose or it's a wrong setting. So NuGet won't break customers in this situation.
+
+### Scenarios from the developer side
+**1. Visual Studio for Mac** 
+
+  It's in .NET Framework code path so it will not be affected by this change.
```

Comment:
Is this still true?  I thought I heard that VS 2022 for Mac now runs on .NET (Core?), and shells out to mono for non-SDK style projects?

Unfortunately this would be bad for us, since that would mean a whole lot more customers have nuget.config files in the legacy path.